R/fetch.R

```diff
@@ -35,25 +39,45 @@ git_fetch <- function(remote = NULL, refspec = NULL, password = askpass,
 #' @rdname fetch
 #' @useDynLib gert R_git_remote_push
 git_push <- function(remote = NULL, refspec = NULL, password = askpass,
-                     ssh_key = NULL, verbose = interactive(), repo = '.'){
+                     ssh_key = NULL, mirror = FALSE, force = FALSE,
+                     verbose = interactive(), repo = '.'){
   repo <- git_open(repo)
   info <- git_info(repo)
+
   if(!length(remote))
     remote <- info$remote
+
   remote <- as.character(remote)
+
   if(!length(remote) || is.na(remote))
     stop("No remote is set for this branch")
+
+  if(isTRUE(mirror)) {
+    refs <- info$reflist
+    # Ignore github's special refs
+    refs <- refs[!grepl("^refs/pull", refs)]
+    refspec <- paste0(refs, ":", refs)
+  }
   if(!length(refspec))
     refspec <- info$head
   refspec <- as.character(refspec)
+  if(isTRUE(force)) {
+    refspec = paste0("+",refspec)
```

Comment:
I don't think this works if refspec == NULL ?

Reply:
Shouldn't that case get picked up by 
```
if(!length(refspec))
    refspec <- info$head
```
is there a case where `info$head` would be `NULL` or `NA`?

Reply:
You can combine the `paste0` and `sub` in one step I think like so: `sub("^\\+?", "+", refspec)`